src/api/java/com/minecolonies/api/tileentities/TileEntityColonyBuilding.java

```diff
@@ -454,16 +454,22 @@ public void markInvDirty()
                 final World world = colony.getWorld();
                 if (world != null)
                 {
-                    //Add additional containers
-                    providers.addAll(building.getAdditionalCountainers().stream()
-                                       .map(world::getTileEntity)
-                                       .collect(Collectors.toSet()));
-                    providers.removeIf(Objects::isNull);
-                    building.getAdditionalCountainers()
-                      .stream()
-                      .map(world::getTileEntity)
-                      .filter(entity -> entity instanceof TileEntityRack)
-                      .forEach(entity -> ((TileEntityRack) entity).setBuildingPos(this.pos));
+                    for (final BlockPos pos : building.getAdditionalCountainers())
+                    {
+                        if (WorldUtil.isBlockLoaded(world, pos))
+                        {
```

Comment:
Don't we need a "dirty" flag on load of the other invs too?

Reply:
ye thats why I added to racks that they notify their building on load

Reply:
kk